Dockerfile

```diff
@@ -0,0 +1,20 @@
+FROM node:9.10.1-alpine as builder
+RUN apk update && apk upgrade && \
+    apk add --no-cache bash git openssh alpine-sdk python python-dev py-pip build-base
+RUN npm install -g yarn
+ADD package.json .
+RUN npm install
```

Comment:
you should always use yarn for reproducible builds

Reply:
Here you should install deps with yarn so lockfile is properly utilized. I don't care really tool used to running package.json tasks

Reply:
btw. here i am referring to "build" as:
- install dependencies 
- transpile sources etc.

The first phase can fail if you don't use a lockfile.